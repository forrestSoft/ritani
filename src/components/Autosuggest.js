import React from 'react'

import Avatar from './Avatar'
import './Autosuggest.scss'

function handleA(login,f,e){
	e.preventDefault()
}

function Autosuggest(props){
	return(
		<div className="autosuggest">
		{console.log('a',props.loading)}
			
			<span className={props.loading ? 'loading' : null }>
				<input 
					value={props.value} 
					onChange = { props.onChange } 
					placeholder="GitHub Username" 
					className={props.loading ? 'loading' : null }/>
				</span>
			 { (props.open) ? 
			 	<div>
					<ul>
						{ 
							props.users && props.users.map( (item, i) => {
								return (
									<li key={item.login} onClick={props.onClick.bind(this, item.login)} >
										<Avatar url={item.avatar_url} login={item.login} />
										<div>
											<p>{item.login}</p>
											<a 
												onClick={handleA.bind(this, item.login, props.onClick)}
												href={item.html_url}>{item.html_url.replace(/https.+?com/, '')}
											</a>
										</div>
									</li>
								)
							})
						}
					</ul> 
					<span>{`Total users from this query: ${props.total}`}</span> 
				</div> : null }
		</div>
	)
}

export default Autosuggest