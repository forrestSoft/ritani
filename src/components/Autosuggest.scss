.autosuggest{
	padding-top: 1vh;

  ul{
    list-style: none;
    padding-left: 0;
	}

  li {
    display: flex;
  }

  .avatar_wrapper {
    margin-right: 2vh;
  }
}
