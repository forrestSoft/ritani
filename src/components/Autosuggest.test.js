import React from 'react'
import {shallow} from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import Autosuggest from './Autosuggest'
import staticState from '../staticState'

it('render matches snapshot', () => {
  const autosuggest = shallowToJson(shallow(<Autosuggest users={staticState.slice(0,3)} open={true} onClick={()=>{}} total={10} />)) 
  expect(autosuggest).toMatchSnapshot();
});

it('calls passed onChange on "onChange"', () => {
	const mock = jest.fn().mockReturnValue('default')
	const autosuggest = shallow(<Autosuggest onChange={mock}/>)
	autosuggest.find('input').simulate('change')
	expect(mock).toBeCalled()
})

it('calls passed onClick on "onClick"', () => {
	const mock = jest.fn().mockReturnValue('default')
	const autosuggest = shallow(<Autosuggest open={true} onClick={mock} users={[{login:'boo', html_url: 'https://github.com/blah'}]} />)
	autosuggest.find('input').simulate('change')
	autosuggest.find('li').simulate('click')
	expect(mock).toBeCalled()
})
