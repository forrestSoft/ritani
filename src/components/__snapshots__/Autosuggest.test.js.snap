// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<div
  className="autosuggest"
>
  <section
    className={null}
  >
    <input
      className={null}
      placeholder="GitHub Username"
    />
  </section>
  <nav>
    <ul>
      <li
        key="myhduck"
        onClick={[Function]}
      >
        <Avatar
          login="myhduck"
          url="https://avatars2.githubusercontent.com/u/1555350?v=4"
        />
        <div>
          <h4>
            myhduck
          </h4>
          <a
            href="https://github.com/myhduck"
            onClick={[Function]}
          >
            .
            /myhduck
          </a>
        </div>
      </li>
      <li
        key="trevor"
        onClick={[Function]}
      >
        <Avatar
          login="trevor"
          url="https://avatars0.githubusercontent.com/u/5945?v=4"
        />
        <div>
          <h4>
            trevor
          </h4>
          <a
            href="https://github.com/trevor"
            onClick={[Function]}
          >
            .
            /trevor
          </a>
        </div>
      </li>
      <li
        key="ogijun"
        onClick={[Function]}
      >
        <Avatar
          login="ogijun"
          url="https://avatars0.githubusercontent.com/u/8826?v=4"
        />
        <div>
          <h4>
            ogijun
          </h4>
          <a
            href="https://github.com/ogijun"
            onClick={[Function]}
          >
            .
            /ogijun
          </a>
        </div>
      </li>
    </ul>
    <aside>
      Total users from this query: 10
    </aside>
  </nav>
</div>
`;
